<launch>
  <arg name="RGB_CAMERA_INFO" default="/camera/rgb/camera_info" />
  <arg name="RGB_IMAGE" default="/camera/rgb/image_rect_color" />
  <arg name="RGB_IMAGE_REMOTE" default="/camera_remote/rgb/image_rect_color" />

  <arg name="DEPTH_CAMERA_INFO" default="/camera/depth_registered/hw_registered/camera_info" />
  <arg name="DEPTH_IMAGE" default="/camera/depth_registered/hw_registered/image_rect"  />
  <arg name="DEPTH_IMAGE_REMOTE" default="/camera_remote/depth_registered/image_rect" />

  <arg name="POINT_CLOUD_REMOTE" value="/camera_remote/depth_registered/points" />

  <arg name="MANAGER" value="astra_decompress" />
  <node name="$(arg MANAGER)" pkg="nodelet" type="nodelet" args="manager"/>

  <node name="rgb_decompress" pkg="image_transport" type="republish"
        args="compressed in:=$(arg RGB_IMAGE) out:=$(arg RGB_IMAGE_REMOTE)" />
  <node name="depth_decompress" pkg="image_transport" type="republish"
        args="compressedDepth in:=$(arg DEPTH_IMAGE) out:=$(arg DEPTH_IMAGE_REMOTE)" />
  <node pkg="nodelet" type="nodelet" name="point_cloud_xyzrgb"
        args="load depth_image_proc/point_cloud_xyzrgb $(arg MANAGER)" output="screen" >
    <remap from="rgb/camera_info" to="$(arg RGB_CAMERA_INFO)" />
    <remap from="rgb/image_rect_color" to="$(arg RGB_IMAGE_REMOTE)" />
    <remap from="depth_registered/image_rect" to="$(arg DEPTH_IMAGE_REMOTE)" />
    <remap from="depth_registered/points" to="$(arg POINT_CLOUD_REMOTE)" />
    <rosparam>
      queue_size: 100
    </rosparam>
  </node>
 
</launch>
